core/src/main/java/io/syndesis/core/Tokens.java

```diff
@@ -106,25 +119,24 @@ public static String fetchProviderTokenFromKeycloak(String providerId) {
             );
         }
 
-        String contentTypeHeader = response.getHeaderString("Content-Type");
-        if (!"application/json".equals(contentTypeHeader)) {
-            throw new IllegalStateException(
-                String.format(
-                    "Unable to retrieve token for provider %s from URL %s, expected Content-Type application/json, received %s",
-                    providerId,
-                    tokenEndpointUrl,
-                    contentTypeHeader
-                )
-            );
+        switch (provider) {
+            case OPENSHIFT:
+                ObjectMapper om = new ObjectMapper();
+                try {
+                    AccessTokenResponse accessToken = om.readValue(responseBody, AccessTokenResponse.class);
+                    return accessToken.getToken();
+                } catch (IOException e) {
+                    throw SyndesisServerException.launderThrowable(e);
+                }
+            case GITHUB:
+                Optional<NameValuePair> accessToken = URLEncodedUtils.parse(responseBody, StandardCharsets.UTF_8)
+                    .stream()
+                    .filter((nameValuePair) -> nameValuePair.getName().equalsIgnoreCase("access_token")).findFirst();
+                return accessToken.orElseThrow(IllegalStateException::new).getValue();
+            default:
+                throw new IllegalArgumentException("Unknown provider type: " + provider);
```

Comment:
nit: would call it "Unsupported" as it is known because we have an enum for it. Also, why not putting the extraction directly on to the enum ? would avoid the switch and might it make clearer.